tools/wake/main.cpp

```diff
@@ -228,6 +227,18 @@ void inspect_database(const CommandLineOptions &clo, Database &db, const std::st
   describe(matching_jobs, get_describe_policy(clo), db);
 }
 
+void inspect_database(const CommandLineOptions &clo, Database &db) {
+  if (clo.tagdag) {
+    JAST json = create_tagdag(db, clo.tagdag);
+    std::cout << json << std::endl;
+    return;
+  } else if (clo.history) {
+    query_runs(db);
+  } else {
+    query_jobs(clo, db);
+  }
+}
```

Comment:
The tagdag printing out here but the two queries embedding their printing internally feels a bit off from a functional point of view -- my instincts are saying that `inspect_database` should be either a thin dispatcher (and the tagdag printing should be done inside `create_tagdag`) or that the `query_*` should be pure DB (and `inspect_database` would handle the printing for all three).  This isn't functional, though, so it's a bit of an overreaction.

Whether you do anything with that or not, the `return` isn't necessary any more with the `else` structure.

Reply:
Got rid of the return statement, wasn't too sure about printing inside `create_tagdag` as the name suggests to return a tagdag, so would have to change the name as well (also dont know if this function will be useful later or not). But if you have strong opinions, I can definitely change the name and make it print inside the function.

Reply:
Looks like @colinschmidt  agrees with printing inside the function, so will go ahead and do that